### PR TITLE
[KYUUBI #3934] Compatiable with Trino rest dto

### LINKED ISSUE #3934: [Subtask] Compatiable with Trino rest dto
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

see trino-client module

### Parent issue

#3901

### Are you willing to submit PR?

- [ ] Yes. I can submit a PR independently to improve.
- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by yehere**
### _Why are the changes needed?_

close #3934 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
Approve for ci run. Btw, put the reason or description after `Why are the changes needed?` next time.

**Comment by yehere** [comment]
> bowenliang123

Ok, The relevant issue #3934  was mentioned above. Now there are some code format problems in the ci

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/trino/api/v1/KyuubiTrinoQueryResultAdaptSuite.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    val metadataResp = fe.be.getResultSetMetadata(opHandle)
+    val tRowSet = fe.be.fetchResults(opHandle, FetchOrientation.FETCH_NEXT, 1000, false)
+    val status = fe.be.getOperationStatus(opHandle)
+
+    val uri = new URI("sfdsfsdfdsf")
+    val results = KyuubiTrinoQueryResultAdapt
+      .createQueryResults("/xdfd/xdf", uri, uri, status, Option(metadataResp), Option(tRowSet))
+
+    print(results.toString)
```
Comment (iodone) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/server/trino/api/v1/KyuubiTrinoQueryResultAdaptSuite.scala:50]: remove

**Comment by yehere** [comment]
@ulysses-you Can you review the code for me

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala [hunk 2]
```diff
@@ -166,4 +175,219 @@ object TrinoContext {
...
+    if (rowSet.getColumns == null) {
+      return rowSet.getRows.asScala
+        .map(t => t.getColVals.asScala.map(v => v.getFieldValue.asInstanceOf[Object]).asJava)
+        .asJava
+    }
+
+    rowSet.getColumns.asScala.foreach {
+      case tColumn if tColumn.isSetBoolVal =>
+        val nulls = util.BitSet.valueOf(tColumn.getBoolVal.getNulls)
+        if (dataSet.isEmpty) {
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala:231]: shall we do initialize at beginning ?
`val resultDataSet = new java.util.ArrayList[java.util.ArrayList](rowSet.getRowsSize)`

**Review by ulysses-you — COMMENTED** [review]
thank you @yehere for the work!

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala (hunk 2 above)
Comment (yehere) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala:231]: Good idea

**Comment by yehere** [comment]
> thank you @yehere for the work!

Please check it for me again

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala [hunk 3]
```diff
@@ -166,4 +175,176 @@ object TrinoContext {
...
+      case tColumn if tColumn.isSetBoolVal =>
+        val nulls = util.BitSet.valueOf(tColumn.getBoolVal.getNulls)
+        if (dataResult.isEmpty) {
+          (1 to tColumn.getBoolVal.getValuesSize).foreach(_ =>
+            dataResult.add(new util.LinkedList[Object]()))
+        }
+
+        tColumn.getBoolVal.getValues.asScala.zipWithIndex.foreach {
+          case (_, rowIdx) if nulls.get(rowIdx) =>
+            dataResult.get(rowIdx).add(None)
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala:235]: None -> null ?

Comment (yehere) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala]: I've added some types testing case

**Comment by ulysses-you** [comment]
@yehere seems the failed test is related, can you fix them ?

**Comment by yehere** [comment]
> @yehere seems the failed test is related, can you fix them ?

of course, but it seems a common problem，can you provide me some tips？

**Comment by ulysses-you** [comment]
The error msg:

```
- create trino request context with header
QueryResults{id=/xdfd/xdf, infoUri=sfdsfsdfdsf, partialCancelUri=sfdsfsdfdsf, nextUri=sfdsfsdfdsf, columns=[io.trino.client.Column@7393f8f7], hasData=true, stats=StatementStats{state=FINISHED_STATE, queued=false, scheduled=false, nodes=0, totalSplits=0, queuedSplits=0, runningSplits=0, completedSplits=0, cpuTimeMillis=0, wallTimeMillis=0, queuedTimeMillis=0, elapsedTimeMillis=0, processedRows=0, processedBytes=0, physicalInputBytes=0, peakMemoryBytes=0, spilledBytes=0, rootStage=null}, error=null, updateType=null, updateCount=0}- test convert *** FAILED ***
  results.getColumns().get(0).getType().equals("INT_TYPE") was false (TrinoContextSuite.scala:95)
QueryResults{id=/xdfd/xdf, infoUri=sfdsfsdfdsf, partialCancelUri=sfdsfsdfdsf, nextUri=sfdsfsdfdsf, columns=[io.trino.client.Column@70a5aa93, io.trino.client.Column@599960f, io.trino.client.Column@1a95e204, io.trino.client.Column@39e063e6, io.trino.client.Column@64cca77b, io.trino.client.Column@2dac7c03, io.trino.client.Column@181c2e70, io.trino.client.Column@22d991c6, io.trino.client.Column@23af0d99], hasData=true, stats=StatementStats{state=FINISHED_STATE, queued=false, scheduled=false, nodes=0, totalSplits=0, queuedSplits=0, runningSplits=0, completedSplits=0, cpuTimeMillis=0, wallTimeMillis=0, queuedTimeMillis=0, elapsedTimeMillis=0, processedRows=0, processedBytes=0, physicalInputBytes=0, peakMemoryBytes=0, spilledBytes=0, rootStage=null}, error=null, updateType=null, updateCount=0}- test convert from table
```

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, meging to master

**Pull request closed without merging**